clientapi/auth/password.go

```diff
@@ -48,7 +49,8 @@ func (t *LoginTypePassword) Request() interface{} {
 
 func (t *LoginTypePassword) Login(ctx context.Context, req interface{}) (*Login, *util.JSONResponse) {
 	r := req.(*PasswordRequest)
-	username := r.Username()
+	// Squash username to all lowercase letters
+	username := strings.ToLower(r.Username())
```

Comment:
I'm not really convinced this is the right thing to do in isolation, as I suspect what would happen is something like this:

1. User goes through the `/register` flow as `Mary`
2. Database entry gets created in `account_accounts` etc as `Mary`
3. User goes through the `/login` flow as `Mary`
4. This line changes `Mary` down to `mary`
5. There's no entry `mary` in `account_accounts` as the query is case-sensitive to avoid a table scan
6. Login fails because the user is not found

Reply:
Actually, looking at it, we seem to enforce `validUsernameRegex` on registration, so this may not be a problem after all.

Reply:
The `create-account` binary provided with dendrite, does not enforce the use of all lower case (it also doesn't honour bcrypt_cost in dendrite.yaml).  One user created their account with an upper case letter, which has been in use for some time now and exists in many events with that upper case letter.  Is this going to break federation of rooms?  Is there a way to recover?

Reply:
Filed a proper issue since this is merged #2073